component/alertrules.libsonnet

```
@@ -77,57 +77,66 @@ local predictESStorage = {
   },
 };
 
-local esStorageGroup = {
-  name: 'elasticsearch_node_storage.alerts',
-  rules: [ predictESStorage ],
+local renderRunbookBaseURL(group, baseURL) = {
+  name: group.name,
+  rules: std.map(
+    function(rule)
+      if (
+        std.objectHas(rule, 'annotations') &&
+        std.objectHas(rule.annotations, 'runbook_url') && (
+          std.length(std.findSubstr('[[.RunbookBaseURL]]', rule.annotations.runbook_url)) > 0 ||
+          std.length(std.findSubstr('[[ .RunbookURL ]]', rule.annotations.runbook_url)) > 0
+        )
+      ) then rule {
+        annotations+: {
+          local round1 = std.strReplace(rule.annotations.runbook_url, '[[.RunbookBaseURL]]', baseURL),
+          local round2 = std.strReplace(round1, '[[ .RunbookURL ]]', baseURL),
+          runbook_url: round2,
+        },
+      } else rule,
+    group.rules
+  ),
 };
 
-local esGroups =
-  loadFile('elasticsearch_operator_prometheus_alerts.yaml')[0].groups +
-  loadFile('fluentd_prometheus_alerts.yaml')[0].groups +
-  [
-    if predict_storage_alert.enabled then esStorageGroup,
-  ];
-
-local lokiGroups = loadFile('lokistack_prometheus_alerts.yaml')[0].groups;
-
-local es_prometheus_rules = kube._Object('monitoring.coreos.com/v1', 'PrometheusRule', 'syn-elasticsearch-logging-rules') {
+local prometheus_rules(groups, baseURL) = kube._Object('monitoring.coreos.com/v1', 'PrometheusRule', 'syn-elasticsearch-logging-rules') {
   metadata+: {
     namespace: params.namespace,
   },
   spec: {
     groups: std.filter(
       function(it) it != null,
       [
-        local r = alertpatching.filterPatchRules(
-          g, ignore_alerts, patch_alerts
-        );
-        if std.length(r.rules) > 0 then r
-        for g in esGroups
+        local r = alertpatching.filterPatchRules(g, ignore_alerts, patch_alerts);
+        local s = renderRunbookBaseURL(r, baseURL);
+        if std.length(s.rules) > 0 then s
```

Could be done as follows:

```suggestion
        if std.length(r.rules) > 0 then renderRunbookBaseURL(r, baseURL)
```